ios/brave-ios/Sources/AIChat/Components/Markdown/AttributedString+Markdown.swift

```diff
@@ -0,0 +1,233 @@
+            }
+          case .codeBlock(_):
+            result[range].font = preferredFont.monospaced() //.italic()
+            result[range].foregroundColor = AttributedString.codeBlockForegroundColor
+            result[range].paragraphStyle = AttributedString.codeBlockParagraphStyle
+          case .blockQuote:
+            result[range].font = preferredFont.monospaced()
+
+          case .thematicBreak:
+            break
+
+          case .table(_):
+            break
+
+          case .tableHeaderRow:
+            break
+
+          case .tableRow(_):
+            break
+
+          case .tableCell(_):
+            break
+            
+          @unknown default:
+            break
+          }
+        }
+        
+        previousListId = listId
+        
+        if range.lowerBound != result.startIndex {
+          result.characters.insert(contentsOf: "\n", at: range.lowerBound)
+        }
+      }
+    
+    self = result
+  }
+  
+  func trimmingCharacters(in characterSet: CharacterSet) -> AttributedString {
+    let invertedSet = characterSet.inverted
+    
+    let startRange = self.characters.firstIndex(where: {
+      $0.unicodeScalars.allSatisfy(invertedSet.contains(_:))
+    })
+    
+    let endRange = self.characters.lastIndex(where: {
+      $0.unicodeScalars.allSatisfy(invertedSet.contains(_:))
+    })
+
+    guard let startLocation = startRange, let endLocation = endRange else {
+      return self
+    }
+
+    return AttributedString(self[startLocation...endLocation])
+  }
+  
+  private typealias LinkAttribute = AttributeScopes.FoundationAttributes.LinkAttribute
+  private typealias PresentationAttribute = AttributeScopes.FoundationAttributes.PresentationIntentAttribute
+  private typealias InlinePresentationAttribute = AttributeScopes.FoundationAttributes.InlinePresentationIntentAttribute
+  
+  private enum MarkDownListType {
+    case none
+    case unordered
+    case ordered
+  }
+  
+  private static var defaultParagraphStyle: NSParagraphStyle {
+    let paragraphStyle = NSMutableParagraphStyle()
+    paragraphStyle.paragraphSpacing = 15.0
+    paragraphStyle.minimumLineHeight = 20.0
+    return paragraphStyle
+  }
+  
+  private static var codeBlockParagraphStyle: NSParagraphStyle {
+    let paragraphStyle = NSMutableParagraphStyle()
+    paragraphStyle.firstLineHeadIndent = 20.0 // Indent for the first line
+    paragraphStyle.headIndent = 20.0  // Indent for all lines
+    paragraphStyle.minimumLineHeight = 20.0
+    paragraphStyle.paragraphSpacingBefore = 20.0
+    return paragraphStyle
+  }
+  
+  private static var listParagraphStyle: NSParagraphStyle {
+    let paragraphStyle = NSMutableParagraphStyle()
+    paragraphStyle.tabStops = [NSTextTab(textAlignment: .left, location: 20.0, options: [:])]
+    paragraphStyle.firstLineHeadIndent = 20.0 // Indent for the first line
+    paragraphStyle.headIndent = 20.0  // Indent for all lines
+    paragraphStyle.minimumLineHeight = 20.0
```

Comment:
Lots of hardcoded values in this file, how does this look when the user has larger fonts enabled/accessibility features?

Reply:
All removed as it doesn't actually make a difference. SwiftUI's AttributedString doesn't seem to recognize it anyway.